### PR TITLE
Add function to retrieve registryURL without user credentials.

### PR DESCRIPTION
**Pull request by jpoler**
Add function BaseURL which includes credentials, and SchemeHostPort which doesn't.

Also refactor tests to use anonymous structs.

### CODE REVIEW

**Comment by jpoler** [comment]
@olegshaldybin @wallyqs @eftychis May I haz review?

**Comment by olegshaldybin** [comment]
Should it be left up to client whether to clear credentials or not?

**Comment by jpoler** [comment]
Since we need to hit the registry for metadata during policy checks, the credentials need to be there if it is a private registry (quay.io). Currently the credentials are cleared immediately after policy checks before anything gets sent to JM for persistent storage. So the client can clear credentials, but if the registry is private, we can't check policy and will return an error.

**Comment by jpoler** [comment]
Thoughts for better security would definitely be appreciated.

**Comment by olegshaldybin** [comment]
I meant the client of this library, not APC client.

**Comment by jpoler** [comment]
There is a function `DockerRegistryURL.ClearUserCredentials()` that will clear credentials. `DockerRegistryURL.String()` returns the full registry URL string, and `DockerRegistryURL.StringNoCredentials()` omits credentials.

**Comment by jpoler** [comment]
@olegshaldybin Fixes pushed re our discussion offline.

File: docker/registry_url.go [hunk 1]
```diff
@@ -184,21 +194,32 @@ func (u *DockerRegistryURL) Path() string {
...
 
-// String returns the full string version of a DockerRegistryURL
-func (u *DockerRegistryURL) String() string {
-	schemeHostPort := u.SchemeHostPort()
-	if schemeHostPort == "" {
-		return ""
+// baseString is wrapped by String and StringNoCredentials.
+// It exposes a flag to indicate whether to include credentials in the BaseURL.
+func (u *DockerRegistryURL) baseString(includeCredentials bool) string {
+	base := u.BaseURLNoCredentials()
```
Comment (olegshaldybin) [inline docker/registry_url.go]: very minor, but it does double work if includeCredentials is true, can put this into else clause.

**Comment by olegshaldybin** [comment]
LGTM

File: docker/registry_url.go (hunk 1 above)
Comment (jpoler) [inline docker/registry_url.go]: fixed here and in baseString. Still LGTY?

**Pull request merged**